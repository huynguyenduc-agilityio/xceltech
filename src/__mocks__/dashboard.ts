export const mockPaySlipData = [
  {
    type: 'Basic Wage',
    amount: '150,000',
    deductions: '-30,000',
    total: '120,000',
  },
  { type: 'Tax', amount: '15,000', deductions: '-3,000', total: '12,000' },
  { type: 'Pension', amount: '15,000', deductions: '-3,000', total: '12,000' },
  {
    type: 'Total Earnings',
    amount: '150,000',
    deductions: '-36,000',
    total: '114,000',
  },
];
