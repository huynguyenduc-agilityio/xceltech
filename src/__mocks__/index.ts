export * from './number';
export * from './dashboard';
