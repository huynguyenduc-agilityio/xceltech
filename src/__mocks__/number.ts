export const minMetricPrefixLimit = 1000;

export const decimalPartMocker = {
  secondDigit: 2,
  thirdDigit: 3,
};

export const numberMocker = {
  hundred: 365,
  thousand: 365259,
  million: 365259000,
  negative: -100,
};
