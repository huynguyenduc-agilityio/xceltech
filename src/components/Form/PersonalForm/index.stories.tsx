import type { Meta, StoryObj } from '@storybook/react';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import PersonalForm from '.';

const queryClient = new QueryClient();

const meta = {
  title: 'Components/PersonalForm',
  component: PersonalForm,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  render: () => (
    <QueryClientProvider client={queryClient}>
      <div className="flex items-center justify-center w-screen py-20 bg-blue-light">
        <div className="bg-white w-[988px] rounded-[15px]">
          <div className="p-10">
            <PersonalForm />
          </div>
        </div>
      </div>
    </QueryClientProvider>
  ),
} satisfies Meta<typeof PersonalForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
