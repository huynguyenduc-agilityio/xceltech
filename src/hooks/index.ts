export * from './useToast';
export * from './useConfirm';
export * from './useAuth';
export * from './useUser';
export * from './useLeave';
export * from './useAccount';
export * from './usePagination';
export * from './useNotification';
export * from './useJob';
export * from './useDocument';
export * from './useEducation';
export * from './useGuarantor';
export * from './useFamily';
export * from './useFinancial';
