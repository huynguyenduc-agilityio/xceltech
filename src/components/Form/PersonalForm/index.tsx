import { useCallback } from 'react';
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';

// Constants
import { JOB_CATEGORY_OPTIONS, MESSAGES } from '@/constants';

// Types
import { IInfoUser, ToastStatus, UpdateProfileFormValues } from '@/types';

// Hooks
import { useGetJobs, useToast, useUpdateInfoUser } from '@/hooks';

// Utils
import { updateProfileSchema } from '@/utils';

// Components
import {
  Button,
  Form,
  FormControl,
  FormField,
  FormItem,
  FormMessage,
  Label,
  Select,
  ImageUpload,
  TextField,
} from '@/components';

export interface IPersonalForm {
  initialValues?: Partial<IInfoUser>;
  onBack?: () => void;
}

const PersonalForm = ({ initialValues, onBack }: IPersonalForm) => {
  const { toast } = useToast();
  const { jobs = [], isJobsLoading } = useGetJobs();
  const { handleUpdateInfoUser, isUpdateInfoLoading } = useUpdateInfoUser();

  const {
    firstName = '',
    lastName = '',
    avatar,
    job = { department: '', jobCategory: '', id: '' },
  } = initialValues || {};

  const defaultValues = {
    firstName,
    lastName,
    avatar,
    department: job?.department || '',
    jobCategory: job?.jobCategory || '',
    id: job?.id,
  };

  const form = useForm<UpdateProfileFormValues>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    resolver: zodResolver(updateProfileSchema),
    defaultValues,
  });

  const {
    control,
    handleSubmit,
    setValue,
    getValues,
    reset,
    formState: { isValid, isDirty, isSubmitting },
  } = form;

  const onSubmit = async (data: UpdateProfileFormValues) => {
    const selectedJob = jobs.find((job) => job.id === data.id);

    const payload: Partial<IInfoUser> = {
      firstName: data.firstName,
      lastName: data.lastName,
      job: {
        ...(selectedJob && { id: selectedJob.id }),
        department: data.department || '',
        jobCategory: data.jobCategory || '',
      },
      ...(data.avatar &&
        data.avatar instanceof File &&
        typeof data.avatar !== 'string' && { avatar: data.avatar }),
    };

    try {
      await handleUpdateInfoUser(payload);

      toast({
        title: MESSAGES.COMMON.UPDATE_SUCCESS('Profile'),
        status: ToastStatus.Success,
      });

      reset(initialValues);

      onBack?.();
    } catch {
      toast({
        title: MESSAGES.COMMON.UPDATE_FAILED('Profile'),
        status: ToastStatus.Error,
      });

      reset(getValues());
    }
  };

  const handleFileChange = useCallback(
    (file: File) => {
      setValue('avatar', file);
    },
    [setValue],
  );

  return (
    <Form {...form}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="flex justify-center">
          <ImageUpload imageUrl={avatar} onImageChange={handleFileChange} />
        </div>

        <div className="space-y-10 mt-10 rounded-[15px]">
          <div className="grid grid-cols-2 gap-5">
            <FormField
              control={control}
              name="firstName"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="First Name"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your first name"
                  errorMessage={error?.message}
                  {...field}
                />
              )}
            />
            <FormField
              control={control}
              name="lastName"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="Last Name"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your last name"
                  {...field}
                  errorMessage={error?.message}
                />
              )}
            />
          </div>
          <FormField
            control={control}
            name="department"
            render={({ field, fieldState: { error } }) => (
              <TextField
                label="Department"
                variant="primary"
                className="space-y-5"
                labelClassName="text-md"
                inputClassName="h-[68px] px-12 rounded-[15px]"
                placeholder="Enter your last name"
                {...field}
                errorMessage={error?.message}
              />
            )}
          />
          <FormField
            control={control}
            name="id"
            render={({ field }) => (
              <FormItem>
                <Label className="text-md">Job Title</Label>
                <FormControl>
                  <Select
                    option={jobs}
                    placeholder="Select your job title"
                    className="bg-blue-light h-[68px] px-12 rounded-[15px]"
                    isDisable={isJobsLoading}
                    {...field}
                    onChange={field.onChange}
                  />
                </FormControl>
                <FormMessage />
              </FormItem>
            )}
          />
          <FormField
            control={control}
            name="jobCategory"
            render={({ field }) => (
              <FormItem>
                <Label className="text-md">Job Category</Label>
                <FormControl>
                  <Select
                    option={JOB_CATEGORY_OPTIONS}
                    placeholder="Select your job category"
                    className="bg-blue-light h-[68px] px-12 rounded-[15px]"
                    {...field}
                    onChange={field.onChange}
                  />
                </FormControl>
                <FormMessage />
              </FormItem>
            )}
          />
          <div className="w-full h-[70px]">
            <Button
              type="submit"
              className="w-[364px] bg-green-primary"
              isLoading={isUpdateInfoLoading}
              disabled={!isDirty || !isValid || isSubmitting}
            >
              Update
            </Button>
          </div>
        </div>
      </form>
    </Form>
  );
};

export default PersonalForm;
