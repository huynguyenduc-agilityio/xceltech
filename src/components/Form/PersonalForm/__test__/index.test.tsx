import { fireEvent, render, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import PersonalForm from '..';

// Hooks
import { useGetJobs, useToast, useUpdateInfoUser } from '@/hooks';

// Mocks
import { mockDataJobs } from '@/__mocks__';
import userEvent from '@testing-library/user-event';

const queryClient = new QueryClient();
jest.mock('@/hooks', () => ({
  useToast: jest.fn(),
  useUpdateInfoUser: jest.fn(),
  useGetJobs: jest.fn(),
}));

const mockToast = jest.fn();
const mockHandleUpdateInfoUser = jest.fn();
const mockHandleUpdateInfoUserError = jest
  .fn()
  .mockRejectedValue(new Error('Network error'));

const renderComponent = () => {
  return render(
    <QueryClientProvider client={queryClient}>
      <PersonalForm />
    </QueryClientProvider>,
  );
};

describe('PersonalForm Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useToast as jest.Mock).mockReturnValue({ toast: mockToast });
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUser,
    });

    (useGetJobs as jest.Mock).mockReturnValue({
      isJobsLoading: false,
      jobs: mockDataJobs,
    });

    global.URL.createObjectURL = jest.fn(
      () => 'blob:http://localhost/fake-url',
    );
    global.URL.revokeObjectURL = jest.fn();
  });

  it('should match snapshot', () => {
    const container = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('submits form and calls handleUpdateInfoUser', async () => {
    const { getByRole, getByPlaceholderText, container, getByTitle } =
      renderComponent();
    const firstNameInput = getByPlaceholderText('Enter your first name');
    const lastNameInput = getByPlaceholderText('Enter your last name');
    const departmentInput = getByPlaceholderText('Enter your department');
    const jobSelect = container.querySelector(
      'select[name="id"]',
    ) as HTMLSelectElement;
    const jobCategory = container.querySelector(
      'select[name="jobCategory"]',
    ) as HTMLSelectElement;
    const input = getByTitle('Upload Image');
    const file1 = new File(['dummy content'], 'test1.png', {
      type: 'image/png',
    });

    fireEvent.change(input, {
      target: { files: [file1] },
    });

    fireEvent.change(firstNameInput, { target: { value: 'John' } });
    fireEvent.change(lastNameInput, { target: { value: 'Doe' } });
    fireEvent.change(departmentInput, { target: { value: 'IT' } });
    fireEvent.change(jobSelect, {
      target: { value: '1' },
    });
    fireEvent.change(jobCategory, {
      target: { value: 'Full Time' },
    });

    const submitButton = getByRole('button', { name: 'Update' });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockHandleUpdateInfoUser).toHaveBeenCalled();
    });
  });

  it('handles submit error gracefully', async () => {
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUserError,
    });

    const { getByRole, getByPlaceholderText, container } = renderComponent();
    const firstNameInput = getByPlaceholderText('Enter your first name');
    const lastNameInput = getByPlaceholderText('Enter your last name');
    const departmentInput = getByPlaceholderText('Enter your department');
    const jobSelect = container.querySelector(
      'select[name="id"]',
    ) as HTMLSelectElement;
    const jobCategory = container.querySelector(
      'select[name="jobCategory"]',
    ) as HTMLSelectElement;

    fireEvent.change(firstNameInput, { target: { value: 'John' } });
    fireEvent.change(lastNameInput, { target: { value: 'Doe' } });
    fireEvent.change(departmentInput, { target: { value: 'IT' } });
    fireEvent.change(jobSelect, {
      target: { value: '1' },
    });
    fireEvent.change(jobCategory, {
      target: { value: 'Full Time' },
    });

    const submitButton = getByRole('button', { name: 'Update' });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalled();
    });
  });
});
