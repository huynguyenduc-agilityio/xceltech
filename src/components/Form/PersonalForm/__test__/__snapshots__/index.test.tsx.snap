// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalForm Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="flex justify-center"
        >
          <div
            class="flex flex-col items-start gap-4"
          >
            <div
              class="space-y-2 w-full"
            >
              <div
                class="relative flex items-center justify-center w-[196px] h-[196px] rounded-full bg-gray-neutral group overflow-hidden"
              >
                <button
                  class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground rounded-full w-full h-full"
                  title="Button Upload Image"
                  type="button"
                >
                  <svg
                    class="lucide lucide-plus text-white"
                    fill="none"
                    height="40"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="40"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 12h14"
                    />
                    <path
                      d="M12 5v14"
                    />
                  </svg>
                </button>
                <div
                  class="absolute inset-0 cursor-pointer pointer-events-none bg-black-default/20 opacity-0 group-hover:opacity-100 transition-opacity flex items-center justify-center"
                >
                  <svg
                    class="lucide lucide-plus text-white"
                    fill="none"
                    height="40"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="40"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 12h14"
                    />
                    <path
                      d="M12 5v14"
                    />
                  </svg>
                </div>
                <input
                  accept="image/*"
                  autocomplete="off"
                  class="font-productsans w-full px-5 py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[54px] hidden"
                  data-testid="input-element"
                  title="Upload Image"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="space-y-10 mt-10 rounded-[15px]"
        >
          <div
            class="grid grid-cols-2 gap-5"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="First Name-label"
                >
                  First Name
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="First Name-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r1:-form-item"
                  name="firstName"
                  placeholder="Enter your first name"
                  value=""
                />
              </div>
            </div>
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Last Name-label"
                >
                  Last Name
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Last Name-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r2:-form-item"
                  name="lastName"
                  placeholder="Enter your last name"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Department-label"
              >
                Department
              </label>
              <input
                aria-describedby=":r3:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Department-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r3:-form-item"
                name="department"
                placeholder="Enter your last name"
                value=""
              />
            </div>
          </div>
          <div
            class="space-y-2 w-full"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            >
              Job Title
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r5:"
              aria-expanded="false"
              aria-label="Select"
              class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-disabled=""
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              disabled=""
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Select your job title
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              disabled=""
              name="id"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            />
          </div>
          <div
            class="space-y-2 w-full"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            >
              Job Category
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r7:"
              aria-expanded="false"
              aria-label="Select"
              class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Select your job category
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
            <select
              aria-hidden="true"
              name="jobCategory"
              style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
              tabindex="-1"
            >
              <option
                value="Full Time"
              >
                Full Time
              </option>
              <option
                value="Part Time"
              >
                Part Time
              </option>
            </select>
          </div>
          <div
            class="w-full h-[70px]"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white disabled:hover:bg-primary h-[70px] py-6 w-[364px] bg-green-primary"
              disabled=""
              type="submit"
            >
              Update
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="flex justify-center"
      >
        <div
          class="flex flex-col items-start gap-4"
        >
          <div
            class="space-y-2 w-full"
          >
            <div
              class="relative flex items-center justify-center w-[196px] h-[196px] rounded-full bg-gray-neutral group overflow-hidden"
            >
              <button
                class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground rounded-full w-full h-full"
                title="Button Upload Image"
                type="button"
              >
                <svg
                  class="lucide lucide-plus text-white"
                  fill="none"
                  height="40"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="40"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
              <div
                class="absolute inset-0 cursor-pointer pointer-events-none bg-black-default/20 opacity-0 group-hover:opacity-100 transition-opacity flex items-center justify-center"
              >
                <svg
                  class="lucide lucide-plus text-white"
                  fill="none"
                  height="40"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="40"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </div>
              <input
                accept="image/*"
                autocomplete="off"
                class="font-productsans w-full px-5 py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[54px] hidden"
                data-testid="input-element"
                title="Upload Image"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="space-y-10 mt-10 rounded-[15px]"
      >
        <div
          class="grid grid-cols-2 gap-5"
        >
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="First Name-label"
              >
                First Name
              </label>
              <input
                aria-describedby=":r1:-form-item-description"
                aria-invalid="false"
                aria-labelledby="First Name-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r1:-form-item"
                name="firstName"
                placeholder="Enter your first name"
                value=""
              />
            </div>
          </div>
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Last Name-label"
              >
                Last Name
              </label>
              <input
                aria-describedby=":r2:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Last Name-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r2:-form-item"
                name="lastName"
                placeholder="Enter your last name"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="grid w-full gap-2"
        >
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              id="Department-label"
            >
              Department
            </label>
            <input
              aria-describedby=":r3:-form-item-description"
              aria-invalid="false"
              aria-labelledby="Department-label"
              autocomplete="off"
              class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-testid="input-element"
              id=":r3:-form-item"
              name="department"
              placeholder="Enter your last name"
              value=""
            />
          </div>
        </div>
        <div
          class="space-y-2 w-full"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
          >
            Job Title
          </label>
          <button
            aria-autocomplete="none"
            aria-controls="radix-:r5:"
            aria-expanded="false"
            aria-label="Select"
            class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
            data-disabled=""
            data-placeholder=""
            data-state="closed"
            dir="ltr"
            disabled=""
            role="combobox"
            type="button"
          >
            <span
              style="pointer-events: none;"
            >
              Select your job title
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down h-4 w-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <select
            aria-hidden="true"
            disabled=""
            name="id"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          />
        </div>
        <div
          class="space-y-2 w-full"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
          >
            Job Category
          </label>
          <button
            aria-autocomplete="none"
            aria-controls="radix-:r7:"
            aria-expanded="false"
            aria-label="Select"
            class="font-productsans flex w-full items-center justify-between border border-input py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light h-[68px] px-12 rounded-[15px]"
            data-placeholder=""
            data-state="closed"
            dir="ltr"
            role="combobox"
            type="button"
          >
            <span
              style="pointer-events: none;"
            >
              Select your job category
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down h-4 w-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <select
            aria-hidden="true"
            name="jobCategory"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              value="Full Time"
            >
              Full Time
            </option>
            <option
              value="Part Time"
            >
              Part Time
            </option>
          </select>
        </div>
        <div
          class="w-full h-[70px]"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white disabled:hover:bg-primary h-[70px] py-6 w-[364px] bg-green-primary"
            disabled=""
            type="submit"
          >
            Update
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
